tests/initcoder_test.go

```diff
@@ -0,0 +1,184 @@
+			_, err := runTransactionWithCodeSizeAndGas(t, net, MAX_INIT_CODE_SIZE+1, sufficientGas)
+			require.ErrorContains(
+				err,
+				"intrinsic gas too low",
+				"unexpectedly succeeded to create contract with init code larger than MAX_INITCODE_SIZE",
+			)
+		})
+	})
+}
+
+func testForVariant(t *testing.T, net *IntegrationTestNet,
+	contract *contractcreator.Contractcreator, variant variant,
+	gasForContract, wordCost uint64) {
+
+	t.Run("charges depending on the init code size", func(t *testing.T) {
+		require := require.New(t)
+
+		createAndGetCost := func(codeLen uint64) uint64 {
+			receipt := createContractWithCodeLenAndGas(t, net, variant, codeLen, sufficientGas)
+			require.Equal(types.ReceiptStatusSuccessful, receipt.Status, "failed to create contract with code length ", codeLen)
+			return getCreateCost(t, contract, receipt)
+		}
+
+		cost1Word := createAndGetCost(30)
+		cost2Words := createAndGetCost(42)
+		cost3Words := createAndGetCost(90)
+
+		require.Equal(wordCost, cost2Words-cost1Word, "cost difference between 1 and 2")
+		require.Equal(wordCost*2, cost3Words-cost1Word, "cost difference between 1 and 3")
+	})
+
+	t.Run("fails without enough gas", func(t *testing.T) {
+		// 4 for a zero byte, 1 to make it fail.
+		receipt := createContractWithCodeLenAndGas(t, net, variant, 1, gasForContract-wordCost-1)
+		require := require.New(t)
+		require.Equal(types.ReceiptStatusFailed, receipt.Status,
+			"unexpectedly succeeded to create contract without enough gas")
+	})
+
+	t.Run("with max init code size", func(t *testing.T) {
+		receipt := createContractWithCodeLenAndGas(t, net, variant, MAX_INIT_CODE_SIZE, sufficientGas)
+		require := require.New(t)
+		require.Equal(types.ReceiptStatusSuccessful, receipt.Status,
+			"failed to create contract with code length ", MAX_INIT_CODE_SIZE)
+	})
+
+	t.Run("aborts with init code size larger than MAX_INITCODE_SIZE", func(t *testing.T) {
+		receipt := createContractWithCodeLenAndGas(t, net, variant, MAX_INIT_CODE_SIZE+1, sufficientGas)
+		require := require.New(t)
+		require.Equal(types.ReceiptStatusFailed, receipt.Status,
+			"unexpectedly succeeded to create contract with init code length greater than MAX_INITCODE_SIZE")
+	})
+}
+
+func createContractSuccessfully(t *testing.T, net *IntegrationTestNet, variant variant, codeLen, gasLimit uint64) *types.Receipt {
+	receipt := createContractWithCodeLenAndGas(t, net, variant, codeLen, gasLimit)
+	require := require.New(t)
+	require.Equal(types.ReceiptStatusSuccessful, receipt.Status, "failed to create contract with code length ", codeLen)
+	return receipt
+}
+
+func createContractWithCodeLenAndGas(t *testing.T, net *IntegrationTestNet, variant variant, codeLen, gasLimit uint64) *types.Receipt {
+	receipt, err := net.Apply(func(opts *bind.TransactOpts) (*types.Transaction, error) {
+		opts.GasLimit = gasLimit
+		return variant(opts, big.NewInt(int64(codeLen)))
+	})
+	if err != nil {
```

Comment:
nit: mixing require and testing apis

Reply:
did not think it was worth to instantiate a new `require` just for this one simple check.